### PR TITLE
Update smellman fork with DGU commits

### PR DESCRIPTION
**Pull request by kentsanggds**
## What

This PR will update the smellman fork that upgrades the spatial extension to python 3 with DGU commits also upgraded. 

Tests are passing on the dev stack, unfortunately the git action tests are not passing but have been introduced in another branch - https://github.com/alphagov/ckanext-spatial/tree/ckan-2.9, which contains all these commits plus additional commits to add harvest tests, fix git action tests and fix CSW.

## How to review

Ideally you should spin up your own docker ckan dev stack to run the tests manually, otherwise it's probably good to just look over the commits to see if there are any issues in the upgrade code. 

## Reference

https://trello.com/c/hL3cMBZh/2313-check-out-other-forked-ckanext-spatial-python3-upgrade-pr

### CODE REVIEW

**Review by 0atman — CHANGES_REQUESTED** [review]
I've made a few small comments here and there, but code all looks really good, obvious improvements in style over what has come before - great :muscle:!

Can I recommend the following tools as essential for any python project:
`black` `flake8` `isort`
And if you're feeling really good, `mypy`!
All pip installable :ok_hand:

**Comment by kentsanggds** [comment]
Good suggestion with these imports - `black flake8 isort` but as its an open source project and I don't want to impose style of these extensions of which there are a few, and given that I have limited time to do the upgrade work I'll just concentrate on the style issues that have been picked up by reviewers and will slowly improve the code style rather than do a big update as part of this upgrade.

File: ckanext/spatial/harvesters/base.py [hunk 1]
```diff
@@ -24,6 +25,7 @@
 from ckan.lib.helpers import json
 from ckan import logic
 from ckan.lib.navl.validators import not_empty
+from ckan.lib.navl.dictization_functions import validate
 from ckan.lib.search.index import PackageSearchIndex
 from ckanext.harvest.harvesters.base import munge_tag
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply ckanext/spatial/harvesters/base.py:30]: I'll sort out the files that I touch for now

File: ckanext/spatial/harvesters/base.py [hunk 2]
```diff
@@ -835,3 +839,26 @@ def _validate_document(self, document_string, harvest_object, validator=None):
...
+            _, errors = validate(tag, logic.schema.default_tags_schema())
+
+            if not errors:
+                continue
+
+            for key, key_errors in errors.items():
+                for error in key_errors:
+                    self._save_object_error(error, self.obj, 'Validation')
+
+            invalid_tags.append(tag['name'])
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply ckanext/spatial/harvesters/base.py:859]: I think this is what it is trying to do. 
As this was a commit added from previous DGU approved PR, I'll probably leave this as it is.

File: ckanext/spatial/tests/test_validation.py [hunk 3]
```diff
@@ -98,12 +85,30 @@ def test_05_series_fail_iso19139_schema(self):
...
         )
-        assert len(errors) > 0
-        assert_in("(gmx.xsd)", errors)
-        assert_in(
-            "'{http://www.isotc211.org/2005/gmd}nosuchelement': This element is not expected.",
-            errors,
-        )
+
+    def assert_passes_all_gemini2_validation(self, xml_filepath, gemini_schematron=validation.Gemini2Schematron):
+        errs = self.get_validation_errors(validation.ISO19139EdenSchema,
```
[the remark this answers is not in the document]
Comment (kentsanggds) [reply ckanext/spatial/tests/test_validation.py:90]: ok, will update this for consistency

**Comment by kentsanggds** [comment]
Closing this PR as superseded by this PR - https://github.com/alphagov/ckanext-spatial/pull/28

**Pull request closed without merging**